draft-ietf-quic-qpack.md

```diff
@@ -829,13 +847,21 @@ represented as an 8-bit prefix string literal.
 {: title="Literal Header Field Without Name Reference"}
 
 
-# Error Handling
+# Error Handling {#error-handling}
 
-The following error code is defined for HTTP/QUIC to indicate all failures of
+The following error codes are defined for HTTP/QUIC to indicate failures of
 QPACK which prevent the stream or connection from continuing:
 
-HTTP_QPACK_DECOMPRESSION_FAILED (0x06):
-: QPACK failed to decompress a frame and cannot continue.
+HTTP_QPACK_DECOMPRESSION_FAILED (TBD):
+: The decoder failed to interpret an instruction on a request or push stream and
+  is not able to continue decoding that header block.  This situation MAY be
```

Comment:
I would like to allow decoder implementations to close the connection even if technically closing the stream would be enough.  This is because stream errors are more easy to hide by retry logic that some products have.  Really, a well-behaved encoder implementation should never cause a decoder to signal an error, so if it does, that should be surfaced as much as possible.

Reply:
I think this is already the intent, but I don't think it hurts to spell it out here.  Also, maybe we can get rid of the word situation in all three error descriptions?

Reply:
Did you mean to remove the error code value?

Reply:
Regarding stream vs connection errors, let me reword this section and every non-reference to stream error elsewhere in the document to flow more smoothly.

Regarding removing the error code, yes, it was intentional.  My understanding is that this is the registry at https://quicwg.org/base-drafts/draft-ietf-quic-http.html#rfc.section.10.5, where currently the next available error code is 0x0012, so this would need to be renumbered (by the RFC editor if the two documents get in the queue at the same time?) anyway.

Reply:
I can get rid of the word "situation" three times by removing those three sentences.  They are covered by other parts of the document anyway.  Also, neither https://quicwg.org/base-drafts/draft-ietf-quic-http.html#rfc.section.6.1 nor https://httpwg.org/specs/rfc7540.html#ErrorCodes bothers to spell out which error codes are for stream errors, which ones for connection errors, and which could be used for both.  So why should we?